modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -1027,11 +1027,13 @@ public String toApiVarName(String name) {
 
     @Override
     public void addImportsToOneOfInterface(List<Map<String, String>> imports) {
-        for (String i : Arrays.asList("JsonSubTypes", "JsonTypeInfo")) {
-            Map<String, String> oneImport = new HashMap<>();
-            oneImport.put("import", importMapping.get(i));
-            if (!imports.contains(oneImport)) {
-                imports.add(oneImport);
+        if (additionalProperties.containsKey(JACKSON)) {
+            for (String i : Arrays.asList("JsonSubTypes", "JsonTypeInfo")) {
+                Map<String, String> oneImport = new HashMap<>();
+                oneImport.put("import", importMapping.get(i));
+                if (!imports.contains(oneImport)) {
+                    imports.add(oneImport);
```

Comment:
this sounds like `imports` should be a set

Reply:
@agilob, Set inside Map is not reliable. Any Map modifications will cause Set not to work properly.
```java
String key = "Wow";
String value = "hello";
HashSet<Map<String, String>> set = new HashSet<>() {{
    add(new HashMap<>());
}};
((HashMap<String,String>)set.toArray()[0]).put(key, value);
boolean result = set.add(new HashMap<>() {{
    put(key, value);
}});
System.err.println(result);
```